src/OrchardCore/Orchard.Environment.Extensions/ExtensionManager.cs

```diff
@@ -90,6 +90,11 @@ public class ExtensionManager : IExtensionManager
 
         public IExtensionInfo GetExtension(string extensionId)
         {
+            if (string.IsNullOrEmpty(extensionId))
```

Comment:
OK, this is what I did to test it using Orchard code base:

Orchard.Cms.Web and Orchard.Users - I referenced 'Telerik.UI.for.AspNet.Core' (or Telerik.UI.for.AspNet.Core'.Trial) NuGet package.

In Orchard.Cms.Web\Startup.cs, ConfigureServices, I added
services.AddKendo();

In OrchardCore.Themes\TheAdmin\Views\Layout.cshtml, added kendo scripts and css:

<script asp-name="jQuery" use-cdn="true" at="Head"></script> <script asp-name="kendo-jszip" use-cdn="true" at="Head"></script> <script asp-name="kendo-all-min" use-cdn="true" at="Head"></script> <script asp-name="kendo_aspnetmvc_min" use-cdn="true" at="Head"></script>
In OrchardCore.Modules\Orchard.Users\Views\Admin\Index.cshtml, I added a dummy grid:
@(Html.Kendo().Grid<TelerikAspNetCoreApp2.Models.OrderViewModel>()
.Name("grid")
.Columns(columns =>
{
columns.Bound(p => p.OrderID).Filterable(false);
columns.Bound(p => p.Freight);
columns.Bound(p => p.OrderDate).Format("{0:MM/dd/yyyy}");
columns.Bound(p => p.ShipName);
columns.Bound(p => p.ShipCity);
columns.Command(command => { command.Edit(); command.Destroy(); }).Width(160);
})
.ToolBar(toolbar => toolbar.Create())
.Editable(editable => editable.Mode(GridEditMode.PopUp))
.Pageable()
.Sortable()
.Scrollable()
.Filterable()
.HtmlAttributes(new { style = "height:550px;" })
.DataSource(dataSource => dataSource
.Ajax()
.PageSize(20)
.Read(read => read.Action("Orders_Read", "Admin"))
.Update(read => read.Action("Orders_Read", "Admin"))
.Create(read => read.Action("Orders_Read", "Admin"))
.Destroy(read => read.Action("Orders_Read", "Admin"))
.ServerOperation(true)
)
)

In Orchard.Users\Controllers\AdminController.cs, added Orders_Read method:
public ActionResult Orders_Read([DataSourceRequest]DataSourceRequest request)
{
var result = Enumerable.Range(0, 50).Select(i => new OrderViewModel
{
OrderID = i,
Freight = i * 10,
OrderDate = new DateTime(2016, 9, 15).AddDays(i % 7),
ShipName = "ShipName " + i,
ShipCity = "ShipCity " + i
});

    var dsResult = result.ToDataSourceResult(request);
    return Json(dsResult);
}
Added OrderViewModel.cs:
using System;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace Orchard.Users.ViewModels
{
[JsonObject(MemberSerialization.OptIn)]
public class OrderViewModel
{
[JsonProperty("OrderID")]
public int OrderID
{
get;
set;
}

    [JsonProperty("Freight")]
    public decimal? Freight
    {
        get;
        set;
    }

    [Required]
    [JsonProperty("OrderDate")]
    public DateTime? OrderDate
    {
        get;
        set;
    }

    [JsonProperty("ShipCity")]
    public string ShipCity
    {
        get;
        set;
    }

    [JsonProperty("ShipName")]
    public string ShipName
    {
        get;
        set;
    }
}
}

I put a grid directly into existing Orchard module to make sure this is really an Orchard issue. This is obviously just a test case. In our code we use custom modules, and it also fails.

You also need to send AntForgeryToken back, so it needs to be like this:

.Read(read => read.Action("Orders_Read", "Admin").Data("kendo.antiForgeryTokens"))
.Update(read => read.Action("Orders_Read", "Admin").Data("kendo.antiForgeryTokens"))
.Create(read => read.Action("Orders_Read", "Admin").Data("kendo.antiForgeryTokens"))
.Destroy(read => read.Action("Orders_Read", "Admin").Data("kendo.antiForgeryTokens"))

although in our case it probably does not matter. If I remember correctly, the failure happens NOT on the call to AJAX method, but rather on rendering Razor view in the first place. If you have CRUD actions defined in the view, I believe Orchard watches the response stream, and tries to do some procession based on the content if the Response. That's where it invokes the code in question that fails.